subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/TestFile.java

```diff
@@ -868,6 +871,19 @@ public URI relativizeFrom(TestFile baseDir) {
         return baseDir.toURI().relativize(toURI());
     }
 
+    /**
+     * Returns a human-readable relative path to this file from the base directory passed to create this TestFile.
+     *
+     * Fails if this TestFile was created in a way that did not provide a relative base.
+     *
+     * @see #relativizeFrom(TestFile)
+     * @see java.nio.file.Path#relativize(Path)
+     */
+    public String getRelativePathFromBase() {
+        Preconditions.checkArgument(!relativeBase.toPath().equals(this.toPath()), "relativeBase must have been set during construction");
```

Comment:
I'm opposed to expanding the use of guava in our codebase. Since this is just a test, it's probably fine and this is not my hill to die on!

But also since this is a test, maybe a JUnit assertion is a better fit here?